internal/test/must.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/random"
+)
+
+// test.T(t).Digest(img)
+//
+// or
+//
+// must := must.T(t)
+// must.Digest(img)
+
+// func TestFoo(t *testing.T) {
+//   must := test.T(t)
+//   img := must.RandomImage(1024, 5)
+//   must.Digest(img)
+//   must.ParseReference("gcr.io/foo/bar")
+//   must.Tag("gcr.io/foo/bar:baz")
+// }
+
+func T(t *testing.T) Muster {
+	return testMuster{t}
+}
+
+type digester interface {
+	Digest() (v1.Hash, error)
+}
+
+type Muster interface {
+	RandomImage(int64, int64) v1.Image
```

Comment:
I'd rather not take a dep on `random` if we can avoid it.

Can this be `Image(v1.Image, error) v1.Image` called as `must.Image(random.Image(...))`?

Reply:
Great idea. Done!